functions/nodejs-apps.bash

```diff
@@ -202,3 +202,147 @@ nodered_setup() {
+    if ! mqttUser=$(whiptail --title "MQTT User" --inputbox "$mqttUserText" 10 80 "$mqttDefaultUser" 3>&1 1>&2 2>&3); then return 0; fi
+    if ! mqttPW=$(whiptail --title "MQTT password" --passwordbox "$mqttPWText" 10 80 3>&1 1>&2 2>&3); then return 0; fi
+    if ! (whiptail --title "Zigbee2MQTT installation" --infobox "$installText" 14 80); then echo "CANCELED"; return 0; fi
+  fi
+  if [[ -z $mqttUser ]] ; then mqttUser="$mqttDefaultUser"; fi
+
+  if ! node_is_installed || is_armv6l; then
+    echo -n "$(timestamp) [openHABian] Installing zigbee2mqtt prerequsites (NodeJS)... "
+    if cond_redirect nodejs_setup; then true; else echo "FAILED"; return 1; fi
+  fi
+
+  echo -n "$(timestamp) [openHABian] Downloading zigbee2mqtt... "
+  zigbee2mqttBase="$(npm list | head -n 1)/node_modules/zigbee2mqtt"
+  if [[ -d $zigbee2mqttBase ]]; then
+    if cond_redirect systemctl stop zigbee2mqtt.service; then echo "OK (stop service)"; else echo "FAILED (stop service)"; return 1; fi # Stop the service 
+    cond_echo "Removing any old installations..."
+    cond_redirect npm uninstall zigbee2mqtt
+  fi
+  if ! cond_redirect zigbee2mqtt_download "/opt"; then echo "FAILED (download)"; return 1; fi
+  echo "OK"
+
+  echo -n "$(timestamp) [openHABian] Creating log directory... "
+  mkdir  /var/log/zigbee2mqtt || (echo "FAILED (create log-directory)"; return 1)
+  chown openhabian /var/log/zigbee2mqtt || (echo "FAILED (create log-directory)"; return 1)
+  chgrp openhab /var/log/zigbee2mqtt || (echo "FAILED (create log-directory)"; return 1)
+  echo "OK"
+
+  echo -n "$(timestamp) [openHABian] zigbee2mqtt configuration... "
+  cd /opt/zigbee2mqtt || (echo "FAILED (cd)"; return 1)
+  if ! cond_redirect sudo -u "${username:-openhabian}" npm install zigbee2mqtt; then echo "FAILED (npm install)"; return 1; fi
+  if ! (sed -e "s|%adapterName|${adapterArray[$selectedAdapter-1]}|g" -e "s|%user|$mqttUser|g" /opt/openhabian/includes/zigbee2mqtt/configuration.yaml | sudo -u "${username:-openhabian}" tee /opt/zigbee2mqtt/data/configuration.yaml); then echo "FAILED (configuration.yaml file creation)"; return 1; fi
+  if [[ -n $mqttPW ]] ; then (sed -i -e "s|#%password:|password: $mqttPW|g" /opt/zigbee2mqtt/data/configuration.yaml ); fi
+  cd /opt || (echo "FAILED (cd)"; return 1)
+  echo "OK"
+  
+
+  echo -n "$(timestamp) [openHABian] Setting up Zigbee2MQTT service... "
+  if ! cond_redirect install -m 644 "${BASEDIR:-/opt/openhabian}"/includes/zigbee2mqtt/zigbee2mqtt.service /etc/systemd/system/; then echo "FAILED (install service)"; return 1; fi
+  if ! cond_redirect systemctl -q daemon-reload; then echo "FAILED (daemon-reload)"; return 1; fi
+  if ! cond_redirect systemctl enable --now zigbee2mqtt.service; then echo "FAILED (enable service)"; return 1; fi
+  if ! cond_redirect systemctl restart zigbee2mqtt.service; then echo "FAILED (restart service)"; return 1; fi
```

Comment:
Why immediately restart the service, unless needed for configuration updates this will only cause problems as the `--now` flag on enable will ensure that the service is started.

Reply:
ah, didn't know that. thanks